AGENT.md

```diff
@@ -0,0 +1,112 @@
+
+### Testing
+
+- **IMPORTANT**: Run `yarn build` before running any E2E tests to ensure all packages are built
+- `yarn --cwd apps/e2e e2e` - Run all E2E tests
+- `yarn --cwd apps/e2e e2e:client` - Run client-specific E2E tests
+- `yarn --cwd apps/e2e e2e:engine` - Run engine-client E2E tests  
+- `yarn --cwd apps/e2e e2e:indexer` - Run indexer-client E2E tests
+- `yarn --cwd apps/e2e e2e:trigger` - Run trigger-client E2E tests
+
+### Package Management
+
+- `yarn link-local` / `yarn unlink-local` - Link/unlink packages for local development
+- `yarn publish-all` - Clean, build, and publish all packages via Lerna
+- `yarn depcruise:all` - Analyze package dependencies and detect circular dependencies
+
+### Individual Package Scripts
+
+Each package in `packages/` has these common scripts:
+- `yarn build` - Build the specific package
+- `yarn clean` - Clean build artifacts  
+- `yarn dev` - Watch mode for development
+- `yarn lint` - Check linting rules only
+- `yarn lint:fix` - Fix linting issues automatically
+- `yarn typecheck` - Type check without emitting files
+
+## Architecture
+
+### Monorepo Structure
+
+The project follows a monorepo pattern with these core packages:
+
+1. **`@vertex-protocol/client`** - Main entry point that composes all other packages into a unified `VertexClient`
+2. **`@vertex-protocol/engine-client`** - Handles off-chain matching engine communication
+3. **`@vertex-protocol/indexer-client`** - Provides indexer queries for historical data
+4. **`@vertex-protocol/trigger-client`** - Manages trigger service for stop orders
+5. **`@vertex-protocol/contracts`** - Contract utilities, ABIs, and on-chain interactions
+6. **`@vertex-protocol/utils`** - Common utilities including BigNumber.js for decimal math
+
+### Client Architecture
+
+- `VertexClient` is the main class that orchestrates all API interactions
+- Uses `viem` for Ethereum wallet/provider functionality (migrated from ethers)
+- Supports both chain signers and linked signers for trading
+- Modular API design with separate classes for Market, Spot, Perp, Subaccount, and WebSocket operations
+
+### Key Patterns
+
+- All packages use TypeScript with strict type checking
+- BigDecimal/BigNumber.js for precise decimal calculations
+- EIP-712 signing for off-chain order execution
+- Comprehensive type definitions for all API responses
+- Consistent error handling with custom error classes
+- Viem as the primary Ethereum library dependency
+
+## Test and Verification Sequence
+
+After making edits, **ALWAYS** run the following verification sequence:
+
+1. **Type Check**  
+   - Run `yarn typecheck` to verify all TypeScript types are correct across all packages.
+2. **Lint Check**  
+   - Run `yarn lint` to run ESLint with auto-fix and Prettier formatting.
+3. **Build**  
+   - Run `yarn build` to build all packages before running any tests.
+4. **Tests**
+   - Run `yarn test` to run all tests across the codebase.
+5. **E2E Tests**
```

Comment:
Hmm, running the E2E suite would take too long if it's doing the sequence multiple times. 

This could be something we do manually as a final manual verification?

Reply:
Sure, I have removed E2E Tests from here.